dartsim/src/EntityManagementFeatures.cc

```diff
@@ -602,6 +658,50 @@ bool EntityManagementFeatures::ModelRemoved(const Identity &_modelID) const
   return !this->models.HasEntity(_modelID);
 }
 
+/////////////////////////////////////////////////
+bool EntityManagementFeatures::RemoveNestedModelByIndex(
+    const Identity &_modelID, std::size_t _modelIndex)
+{
+  auto modelInfo = this->ReferenceInterface<ModelInfo>(_modelID);
+  if (_modelIndex >= modelInfo->nestedModels.size())
+  {
+    return this->GenerateInvalidId();
+  }
+  const auto nestedModelID = modelInfo->nestedModels[_modelIndex];
+  if (this->models.HasEntity(nestedModelID))
+  {
+    auto worldID = this->GetWorldOfModelImpl(nestedModelID);
+    auto model = this->models.at(nestedModelID)->model;
+    auto filterPtr = GetFilterPtr(this, worldID);
+    filterPtr->RemoveSkeletonCollisions(model);
+    return this->RemoveModelImpl(worldID, nestedModelID);
+  }
+  return false;
+}
+
+/////////////////////////////////////////////////
+bool EntityManagementFeatures::RemoveNestedModelByName(const Identity &_modelID,
+                                                 const std::string &_modelName)
+{
+  auto modelInfo = this->ReferenceInterface<ModelInfo>(_modelID);
+  const std::string fullName =
+      ::sdf::JoinName(modelInfo->model->getName(), _modelName);
+
+  if (this->models.HasEntity(_modelID))
+  {
+    auto worldID = this->GetWorldOfModelImpl(_modelID);
+    auto nestedSkel = this->worlds.at(worldID)->getSkeleton(fullName);
+    if (nullptr == nestedSkel || !this->models.HasEntity(nestedSkel))
+    {
+      return false;
+    }
+    const std::size_t nestedModelID = this->models.IdentityOf(nestedSkel);
+    auto filterPtr = GetFilterPtr(this, worldID);
```

Comment:
Nitpick: Since we never change what `filterPtr` is pointing at, we can add const here.

```suggestion
    const auto filterPtr = GetFilterPtr(this, worldID);
```

Reply:
https://github.com/ignitionrobotics/ign-physics/pull/230/commits/62abc8babaa68db286f0ab24657042f83c7504c6